wordpress-mobile/WordPress-FluxC-Android
# [Coupons] Add suspended functions and observed DB for products, variations and categories

This PR adds the product, variation and category search & fetch suspended functions required for the product/category selection dialog in coupon editing. It also adds the observation functions that return `Flow`s.

**To test:**
Tests the **observation**:
1. Go to Woo -> Products
2. Select Fetch product categories
3. Then select Observe product categories
4. Notice a list of product categories is shown (uses the new observe function)

Tests the new product fetching using parameters:
1. Go to Woo -> Products
2. Select fetch specific products
3. Enter valid product IDs, separated by comma
4. Notice the products with valid IDs get fetched

Also, run the unit tests and make sure they pass.

**Note:** Please merge after #2407.
